internal/consensus/common_test.go

```diff
@@ -122,7 +123,21 @@ func (vs *validatorStub) signVote(
 	}
 	v := vote.ToProto()
 	err = vs.PrivValidator.SignVote(context.Background(), config.ChainID(), v)
+
+	// ref: signVote in FilePV, the vote should use the privious vote info when the sign data is the same.
+	if vs.lastVote != nil &&
```

Comment:
This if check is pretty long and it makes readability a bit difficult because it's hard to tell what its intention is from a glance. Can we wrap this in a helper function. Perhaps `signDataIsEqual(vote, vote) bool` that we can then use in this conditional.

Reply:
ACK. Good idea!

Reply:
sure. will update!